### PR TITLE
Feature/stripe transaction detail

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3332020&stafilter=NotComplete&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Models/Foundation/Summit/Registration/SummitOrder.php [hunk 1]
```diff
@@ -318,6 +330,11 @@ public function setPaid()
             $ticket->setPaid();
         }
 
+        if (!is_null($payload)) {
+            $this->credit_card_type = $payload['order_credit_card_type'];
```
Comment (smarcet) [inline app/Models/Foundation/Summit/Registration/SummitOrder.php:334]: please check that keys exists

File: database/migrations/model/Version20230720172326.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema): void
+    {
+        $builder = new Builder($schema);
+        if($schema->hasTable("SummitOrder")) {
+            $builder->table("SummitOrder", function (Table $table) {
+                $table->string("CreditCardType")->setNotnull(false)->setLength(50);
```
Comment (smarcet) [inline database/migrations/model/Version20230720172326.php:34]: lets default to NULL both fields

File: app/Services/Apis/PaymentGateways/StripeApi.php [hunk 3]
```diff
@@ -181,25 +200,28 @@ public function processCallback(LaravelRequest $request): array
                 throw new \InvalidArgumentException();
 
             $intent = $event->data->object;
+
+            $credircard_info = $this->getCreditCardInfo($intent->toArray());
```
Comment (smarcet) [inline app/Services/Apis/PaymentGateways/StripeApi.php:204]: i do think there is a typo here

File: app/Services/Apis/PaymentGateways/StripeApi.php [hunk 4]
```diff
@@ -86,6 +86,25 @@ public function setWebHookSecretKey(string $webhook_secret_key):void{
         $this->webhook_secret_key = $webhook_secret_key;
     }
 
+    public function getCreditCardInfo(array $payload): array
+    {
+        $charges = $payload['charges']['data'];
```
Comment (smarcet) [inline app/Services/Apis/PaymentGateways/StripeApi.php:91]: check this keys exists at payload before doing evaluation

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: app/Services/Apis/PaymentGateways/StripeApi.php (hunk 3 above)
Comment (romanetar) [reply app/Services/Apis/PaymentGateways/StripeApi.php:204]: typo fixed

File: app/Services/Apis/PaymentGateways/StripeApi.php (hunk 4 above)
Comment (romanetar) [reply app/Services/Apis/PaymentGateways/StripeApi.php:93]: done

**Pull request merged**